pkg/util/conditions/conditions_test.go

```diff
@@ -0,0 +1,196 @@
+	"testing"
+
+	healthcheckingv1alpha1 "github.com/openshift/machine-api-operator/pkg/apis/healthchecking/v1alpha1"
+
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	namespace   = "openshift-cluster-api"
+	correctData = `items:
+- name: Ready 
+  timeout: 60s
+  status: Unknown`
+	unrelatedConditionData = `items:
+- name: Unrelated 
+  timeout: 60s
+  status: Unknown`
+	unrelatedStatusData = `items:
+- name: Ready 
+  timeout: 60s
+  status: Unrelated`
+)
+
+type expectedConditions struct {
+	conditions []UnhealthyCondition
+	error      bool
+}
+
+func node(name string, ready bool) *v1.Node {
+	nodeReadyStatus := corev1.ConditionTrue
+	if !ready {
+		nodeReadyStatus = corev1.ConditionUnknown
+	}
+
+	return &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: metav1.NamespaceNone,
+			Labels:    map[string]string{},
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind: "Node",
+		},
+		Status: corev1.NodeStatus{
+			Conditions: []corev1.NodeCondition{
+				{
+					Type:   corev1.NodeReady,
+					Status: nodeReadyStatus,
+				},
+			},
+		},
+	}
+}
+
+func configMap(name string, data map[string]string) *corev1.ConfigMap {
+	return &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind: "ConfigMap",
+		},
+		Data: data,
+	}
+}
+func TestGetNodeUnhealthyConditions(t *testing.T) {
+	nodeHealthy := node("nodeHealthy", true)
+	nodeRecentlyUnhealthy := node("nodeRecentlyUnhealthy", false)
+
+	cmWithoutConditions := configMap(
+		healthcheckingv1alpha1.ConfigMapNodeUnhealthConditions,
+		map[string]string{
+			"noconditions": "",
+		},
+	)
+	cmWithBadData := configMap(
+		healthcheckingv1alpha1.ConfigMapNodeUnhealthConditions,
+		map[string]string{
+			"conditions": "unparsed-data",
+		},
+	)
+	cmWithUnrelatedCondition := configMap(
+		healthcheckingv1alpha1.ConfigMapNodeUnhealthConditions,
+		map[string]string{
+			"conditions": unrelatedConditionData,
+		},
+	)
+	cmWithUnrelatedConditionStatus := configMap(
+		healthcheckingv1alpha1.ConfigMapNodeUnhealthConditions,
+		map[string]string{
+			"conditions": unrelatedStatusData,
+		},
+	)
+	cmWithCorrectData := configMap(
+		healthcheckingv1alpha1.ConfigMapNodeUnhealthConditions,
+		map[string]string{
+			"conditions": correctData,
+		},
+	)
+
+	testsCases := []struct {
+		name     string
+		node     *corev1.Node
+		cm       *corev1.ConfigMap
+		expected expectedConditions
+	}{
+		{
+			name: "healthy node",
+			node: nodeHealthy,
+			cm:   cmWithCorrectData,
+			expected: expectedConditions{
+				conditions: []UnhealthyCondition{},
+				error:      false,
+			},
+		},
+		{
+			name: "unhealthy node",
+			node: nodeRecentlyUnhealthy,
+			cm:   cmWithCorrectData,
+			expected: expectedConditions{
+				conditions: []UnhealthyCondition{
+					{
+						Name:    "Ready",
+						Timeout: "60s",
+						Status:  "Unknown",
+					},
```

Comment:
why unknown? what about false?

Reply:
I do not sure when the kubelet get `False` condition, in most problematic cases that I saw, kubelet just stop to talk with the master, so the node has `Unknown` status.